packages/test/test-driver-definitions/src/localServerTestDriver.ts

```diff
@@ -0,0 +1,17 @@
+/*!
+ * Copyright (c) Microsoft Corporation. All rights reserved.
+ * Licensed under the MIT License.
+ */
+import { ISummaryConfiguration } from "@fluidframework/protocol-definitions";
+import { ILocalDeltaConnectionServer } from "@fluidframework/server-local-server";
+import { ITestDriver } from "./interfaces";
+
+export interface ILocalServerTestDriver extends Omit<ITestDriver,"type"> {
```

Comment:
Maybe I missed it, but I don't see why we need this specialized interface. Did you mean for container.spec.ts to use the interface type instead of the class type?  The only difference (once `reset` is removed per the comment) is that `server` is exposed, right?

Reply:
i need the different type so i can use a discriminated union which give strong typing when type === "local"

Reply:
Right, I didn't see any usage like that in this PR, but makes sense that it'll be needed.

I don't love that the 4 cases are now defined far apart from each other... but I guess intellisense will tell you what the possibilities are etc.

Another way to do this that to me is more illustrative would be something like this:
```
interface IBaseTestDriver {
  ...All the stuff on ITestDriver except type...
}

type TestDriverFeatures = {
  readonly type: ""tinylicious" | "routerlicious" | "odsp";
} | {
  readonly type: "local";
  readonly server: ILocalDeltaConnectionServer
}

type TestDriver = IBaseTestDriver & TestDriverFeatures
```

Then if one of the other drivers also gets a special feature that you need type narrowing to access it's easy to add.

Totally subjective, your call.

Reply:
in the above i don't understand how we link the local server implementation to the TestDriverFeatures

Reply:
also, i'm thinking the ILocalServerTestDriver might just go away, as i don't see us having tests that optionally need local server to work. we may have a subset of tests that need local server, but those tests won't make sense in any other context, so we can just put them in their own package

Reply:
> in the above i don't understand how we link the local server implementation to the TestDriverFeatures at compile time

Great point 😂 That's what I get for coding in Markdown